File: js/src/bindings/burnDomain.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.burnDomain = void 0;
const web3_js_1 = require("@solana/web3.js");
const burnInstruction_1 = require("../instructions/burnInstruction");
const constants_1 = require("../constants");
const getDomainKeySync_1 = require("../utils/getDomainKeySync");
const getReverseKeySync_1 = require("../utils/getReverseKeySync");
const burnDomain = (domain, owner, target) => {
    const { pubkey } = (0, getDomainKeySync_1.getDomainKeySync)(domain);
    const [state] = web3_js_1.PublicKey.findProgramAddressSync([pubkey.toBuffer()], constants_1.REGISTER_PROGRAM_ID);
    const [resellingState] = web3_js_1.PublicKey.findProgramAddressSync([pubkey.toBuffer(), Uint8Array.from([1, 1])], constants_1.REGISTER_PROGRAM_ID);
    const ix = new burnInstruction_1.burnInstruction().getInstruction(constants_1.REGISTER_PROGRAM_ID, constants_1.NAME_PROGRAM_ID, web3_js_1.SystemProgram.programId, pubkey, (0, getReverseKeySync_1.getReverseKeySync)(domain), resellingState, state, constants_1.REVERSE_LOOKUP_CLASS, owner, target);
    return ix;
};
exports.burnDomain = burnDomain;

File: python/src/bindings/burn_domain.py
from dis import Instruction

from solana.constants import SYSTEM_PROGRAM_ID
from solders.pubkey import Pubkey

from constants import REGISTER_PROGRAM_ID, NAME_PROGRAM_ID, REVERSE_LOOKUP_CLASS
from instructions.BurnInstruction import BurnInstruction
from utils import get_domain_key, get_reverse_key


def burn_domain(
        domain: str,
        owner: Pubkey,
        target: Pubkey,
) -> Instruction:
    pubkey = get_domain_key(domain)["pubkey"]
    state, _ = pubkey.find_program_address(
        [bytes(pubkey)],
        REGISTER_PROGRAM_ID,
    )
    reselling_state, _ = pubkey.find_program_address(
        [bytes(pubkey), bytes([1, 1])],
        REGISTER_PROGRAM_ID,
    )

    return BurnInstruction.get_instruction(
        REGISTER_PROGRAM_ID,
        NAME_PROGRAM_ID,
        SYSTEM_PROGRAM_ID,
        pubkey,
        get_reverse_key(domain),
        reselling_state,
        state,
        REVERSE_LOOKUP_CLASS,
        owner,
        target
    )
